docs/DeveloperGuide.adoc

```diff
@@ -478,10 +478,41 @@ image::UndoRedoActivityDiagram.png[width="200"]
 // end::undoredo[]
 
 // tag::EnhancedAddCommand[]
-=== Enhanced add command mechanism
-A more flexible add command that only requires the name to be specified at the time of its execution. The other fields such as phone, email, address and remark need not be specified at the time of add command execution. +
+=== Enhanced `add` command mechanism
+
+A more flexible add command that only requires the name to be specified at the time of its execution. The other fields such as phone, email, address and remark need not be specified at the time of add command execution but the user can still choose to do so if they desire. +
+
+The Command is still being supported by the `undoableCommand` class and user will be able to undo or redo this command. +
+
+The unspecified field will be substituted by the string "unspecified field". For example, an unspecified phone number that is not typed in during add command execution will result in "unspecified Phone" to be displayed under the newly added person entry. +
+
+Before: all fields needs to be specified before adding is done successfully +
+
+image::addCommandDemo.png[width="800"]
+
+After: Only name is compulsory for a successful add command +
+
+image::EnhancedAddDemo.png[width="800"]
+
+
 A separate parser for phone, email, address and remark are used to accommodate this enhancement to isolate the interdependency between the modified enhanced add command with the other commands. +
 
+An example is the `parsePhone` method which are also shared with `editCommand`, hence modifying the `parsePhone` method will cause a dependency issue for `editCommand` as well since they requires different parsing method after `addCommand` is enhanced. +
+
+Therefore a separate `parseAddPhone` method is created so that such dependency can be avoided. This is also being implemented to all the other non compulsory fields (email, address, remark).
+
+image::ParsePhone.png[width="800"]
+
+
+It works by instead of returning an empty Optional field, example an empty `Phone` object which sets the internal phone strings to null, we used a substitute string "unspecified Phone" instead. +
```

Comment:
What...does this line mean?

Reply:
The empty optional phone object created sets its own phone string attribute to null